src/valkey_test_case.py

```diff
@@ -92,9 +105,32 @@ def set_startup_args(self, args):
         self.args.update(args)
 
     def get_new_client(self):
-        return self.create_from_server(self)
+        if self.external_mode:
+            client = StrictValkey(host=self.bind_ip, port=self.port)
+            self.clients.append(client)
+            return client
+        else:
+            return self.create_from_server(self)
 
     def exit(self, cleanup=True, remove_nodes_conf=True):
+        if self.external_mode:
```

Comment:
We can skip all of this if we just use `create_from_server` in the function above